Backend/ClientActions.php

```diff
@@ -41,10 +41,12 @@ public function getZonesReturnMagentoZone() {
         $response = $this->api->callAPI($this->request);
         if($this->api->responseOk($response)) {
             $magentoZone = null;
+            $bestMatch = strlen($magentoDomainName);
             foreach ($response['result'] as $zone) {
-                if (strstr($magentoDomainName, $zone['name'])) {
+                $firstOccurrence = strpos($magentoDomainName, $zone['name']);
+                if ($firstOccurrence !== false && $firstOccurrence < $bestMatch) {
+                    $bestMatch = $firstOccurrence;
```

Comment:
I doubled checked and realised this case should not happen since if the users `$magentoDomainName` is `domain.com` and `domain.com` is not registered to cloudflare it'll fail regardless. So your assumption is correct.